ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -158,12 +137,67 @@ public void installServices() {
         List<String> services = getConfig(HADOOP_SERVICES);
         List<String> hosts = getAttribute(AMBARI_SERVER).getAttribute(AmbariServer.REGISTERED_HOSTS);
 
-        LOG.debug("About to create cluster with services: " + services);
-        if (services != null && services.size() > 0) {
+        if (isServicesBasedDeployment(services)) {
+            LOG.debug("About to create cluster with services: " + services);
             getAttribute(AMBARI_SERVER).installHDP("Cluster1", "mybp", hosts, services);
         } else {
-            getAttribute(AMBARI_SERVER).installHDP("Cluster1", "mybp", hosts, DEFAULT_SERVICES);
+            AmbariConfig config = ambariConfigFromHostgroups();
+            if (config.hasHostGroups()) {
+                getAttribute(AMBARI_SERVER).installHDPFromConfig("Cluster1", "mybp", config);
+            } else {
+                getAttribute(AMBARI_SERVER).installHDP("Cluster1", "mybp", hosts, DEFAULT_SERVICES);
+            }
+        }
+    }
+
+    private boolean isServicesBasedDeployment(List<String> services) {
+        // As opposed to components based deployment
+        return services != null && services.size() > 0;
+    }
+
+    private AmbariConfig ambariConfigFromHostgroups() {
+        AmbariConfig config = new AmbariConfig();
+        for (Entity entity : getChildren()) {
+            if (entity instanceof AmbariHostGroup) {
+                Collection<Entity> children = entity.getAttribute(AmbariHostGroup.AMBARI_AGENTS).getChildren();
+                config.add(
+                        entity.getDisplayName(),
+                        getHostFQDNs(children),
+                        entity.getConfig(AmbariHostGroup.HADOOP_COMPONENTS)
+                );
+            }
         }
+        return config;
+    }
+
+    private List<String> getHostFQDNs(Collection<Entity> ambariAgents) {
+        ImmutableList.Builder<String> builder = ImmutableList.<String>builder();
+        for (Entity agent : ambariAgents) {
+            if (agent instanceof AmbariAgent) {
+                builder.add(agent.getAttribute(AmbariConfigAndSensors.FQDN));
```

Comment:
Will throw NullPointerException if FQDN is not set on the agent. Not sure if possible but if not now then likely candidate for one that creeps in in the future.